transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -463,6 +467,19 @@ private void processReady(EpollEventArray events, int ready) {
         }
     }
 
+    private void handleDeferredReads() {
+        if (deferEventFdRead) {
+            deferEventFdRead = false;
+            // consume wakeup event.
+            Native.eventFdRead(eventFd.intValue());
```

Comment:
@carl-mastrangelo I wonder if we even need to call eventFdRead(...) at all if we use ET. I think we could just ignore it and only call `eventFdRead(...)` when `eventFdWrite(...)` fails with `EAGAIN` to reset the counter. 

http://man7.org/linux/man-pages/man2/eventfd.2.html

WDYT ?

Reply:
This would also mean we would need to add it with ET:

```
https://github.com/netty/netty/blob/6536b97b5d7fb6847732e0ef5cccf2d00418315e/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java#L103
```

Reply:
I tried this:
```diff
diff --git a/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java b/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java
index cac0f08110..424a4da657 100644
--- a/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java
+++ b/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java
@@ -100,7 +100,7 @@ class EpollEventLoop extends SingleThreadEventLoop {
             this.epollFd = epollFd = Native.newEpollCreate();
             this.eventFd = eventFd = Native.newEventFd();
             try {
-                Native.epollCtlAdd(epollFd.intValue(), eventFd.intValue(), Native.EPOLLIN);
+                Native.epollCtlAdd(epollFd.intValue(), eventFd.intValue(), Native.EPOLLIN | Native.EPOLLET);
             } catch (IOException e) {
                 throw new IllegalStateException("Unable to add eventFd filedescriptor to epoll", e);
             }
@@ -471,7 +471,7 @@ class EpollEventLoop extends SingleThreadEventLoop {
         if (deferEventFdRead) {
             deferEventFdRead = false;
             // consume wakeup event.
-            Native.eventFdRead(eventFd.intValue());
+            //Native.eventFdRead(eventFd.intValue());
         }
         if (deferTimerFdRead) {
             deferTimerFdRead = false;
```

But the Ops/s seems to go down.     (even skipping the re-read logic).    Not sure why yet.

Reply:
Well, maybe just a glitch...    Running with the same change:

```
Benchmark                              Mode  Cnt       Score      Error  Units
EpollSocketChannelBenchmark.execute   thrpt   20  170500.208 ± 2098.283  ops/s
EpollSocketChannelBenchmark.pingPong  thrpt   20   61639.166 ±  416.299  ops/s
```

About the same, and I confirmed with strace the reads aren't happening any more.  I like the idea, but it doesn't seem to affect the `execute` result.   pingPong seems faster, but again, that may be noise.

Lastly, eventFdWrite throws a ChannelException, rather than a specific errno exception, so I am not sure how to programmatically access it.